test/gtest/ucp/test_ucp_am.cc

```diff
@@ -294,13 +298,22 @@ UCS_TEST_P(test_ucp_am, set_am_handler_realloc)
 
 UCS_TEST_P(test_ucp_am, max_am_header)
 {
-    ucp_worker_attr_t attr;
+    size_t min_am_bcopy = SIZE_MAX;
 
+    for (ucp_rsc_index_t idx = 0; idx < sender().ucph()->num_tls; ++idx) {
+        uct_iface_attr_t *attr = ucp_worker_iface_get_attr(sender().worker(), idx);
+        if (attr->cap.flags & UCT_IFACE_FLAG_AM_BCOPY) {
+            min_am_bcopy = ucs_min(min_am_bcopy, attr->cap.am.max_bcopy);
+        }
+    }
+
+    ucp_worker_attr_t attr;
     attr.field_mask = UCP_WORKER_ATTR_FIELD_MAX_AM_HEADER;
 
     ASSERT_UCS_OK(ucp_worker_query(sender().worker(), &attr));
 
     EXPECT_GE(attr.max_am_header, 64ul);
+    EXPECT_LT(attr.max_am_header, min_am_bcopy);
```

Comment:
what if min_am_bcopy is SIZE_MAX in the test?

Reply:
we requested AM feature, so it should not be this way. But if some transport supports `SIZE_MAX` bcopy fragment, we just check that am header is less than that

Reply:
i mean, the test is just duplication of the code itself, should we check that at least one transport was found?